scaleway/resource_k8s_cluster_beta.go

```diff
@@ -709,12 +709,19 @@ func resourceScalewayK8SClusterBetaDefaultPoolUpdate(d *schema.ResourceData, m i
 			if err != nil {
 				return err
 			}
+			defaultPoolID = newRegionalId(region, defaultPoolRes.ID)
 			defaultPool := map[string]interface{}{}
-			defaultPool["pool_id"] = newRegionalId(region, defaultPoolRes.ID)
+			defaultPool["pool_id"] = defaultPoolID
 
 			_ = d.Set("default_pool", []map[string]interface{}{defaultPool})
 
 			if oldPoolID != "" {
+				// wait for new pool to be ready before deleting old one
+				err = waitK8SPoolReady(k8sAPI, region, expandID(defaultPoolID))
+				if err != nil {
+					return err
+				}
+
```

Comment:
@Sh4d1 the problem doing it here, is that TF won't return until the new pool is ready. But I guess there's nothing around it :/

Reply:
Yep but at least it will crash an error is detected :)